tests/csm/rest/test_information.py

```diff
@@ -0,0 +1,168 @@
+
+    @pytest.mark.lc
+    @pytest.mark.parallel
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-42789')
+    def test_42789(self):
+        """
+        Test that user can check version compatability.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+
+        self.log.info("[START] Testing Version Compatability with incompatible version rules")
+        payload = self.csm_obj.get_version_compatibility_payload("incompatible")
+        self.log.info("payload :  %s", payload)
+        response = self.csm_obj.verify_version_compatibility("node", "control-control", payload)
+        res_dict = response.json()
+        assert response.status_code == HTTPStatus.OK, "Status code check failed"
+        assert not res_dict["compatible"], "Compatibility Check failed"
+        success_reason_msg = "Versions are compatible for update."
+        assert res_dict["reason"] != success_reason_msg, "response reason is not correct"
+        self.log.info("[END] Testing Version Compatability with incompatible version rules")
+
+        self.log.info("##### Test ended -  %s #####", test_case_name)
+
+
+    @pytest.mark.lc
+    @pytest.mark.parallel
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-42790')
+    def test_42790(self):
+        """
+        Test that user can check version compatability for invalid payload.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+
+        self.log.info("[START] Testing Version Compatability with invalid resource")
+        # For Non-Happy path with invalid inputs
+        payload = self.csm_obj.get_version_compatibility_payload()
+        self.log.info("payload :  %s", payload)
+        response = self.csm_obj.verify_version_compatibility("cluster", "cortx-cluster", payload)
+        assert response.status_code == HTTPStatus.NOT_FOUND, "Status code check failed"
+
+        self.log.info("[END] Testing Version Compatability with invalid resource")
+        self.log.info("##### Test ended -  %s #####", test_case_name)
+
+    @pytest.mark.lc
+    @pytest.mark.parallel
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-42719')
+    def test_42719(self):
+        """
+        Test that user can check version compatability for invalid payload.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+
+        self.log.info("[START] Testing Version Compatability with invalid rules")
+        payload = self.csm_obj.get_version_compatibility_payload("invalid_rules")
+        self.log.info("payload :  %s", payload)
+        response = self.csm_obj.verify_version_compatibility("node", "cortx-cluster", payload)
+        assert response.status_code == HTTPStatus.INTERNAL_SERVER_ERROR, "Status code check failed"
```

Comment:
please add msg check for negative scenarios
ref: CSM_REST_CFG["msg_check"]

Reply:
updated msg check for negative scenarios